build/gulp/sh.ts

```diff
@@ -1,13 +1,28 @@
-import { exec } from 'child_process'
+import { spawn } from 'child_process'
+
+const sh = (command, cb) => {
+  const [cmd, ...args] = command.split(' ')
 
-const sh = (cmd, cb) => {
   const options = {
     cwd: process.cwd(),
     env: process.env,
-    stdio: 'inherit',
   }
 
-  exec(cmd, options, cb)
+  const child = spawn(cmd, args, options)
+
+  child.stdout.on('data', data => {
+    console.log(data.toString())
+  })
+
+  child.stderr.on('data', data => {
+    console.error(data.toString())
+  })
+
+  child.on('close', code => {
+    if (code === 0) return cb()
+
+    cb(new Error(`child process exited with code ${code}`))
+  })
```

Comment:
Updated shell script helper to print output of the commands it is running.  This is only used in gulp tasks.

Reply:
would recommend to pass exit code to the generated Error somehow - now this information is, essentially, lost down the error processing chain at this point

Reply:
In gulp, calling the callback with an error signifies task failure.  It is usually up to Gulp and any plugins the user has upstream to handle errors.

That said, is there a use case you are thinking of for this error code?